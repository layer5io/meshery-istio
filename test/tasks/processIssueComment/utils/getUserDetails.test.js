const nock = require('nock')

const getUserDetails = require('../../../../src/tasks/processIssueComment/utils/getUserDetails')
const {
    UserNotFoundError,
} = require('../../../../src/tasks/processIssueComment/utils/errors')

const { rejectionOf } = require('../../../testUtils')
const mockUserAPIReturn = require('../../../fixtures/users.getByUsername.jakebolam.json')
const mockGithub = require('../../../mocks/mockGithub')

describe('Get User Details', () => {
    test('gets user details when user exists', async () => {
        const mockUser = 'jakebolam'

        nock('https://api.github.com')
            .get(`/users/${mockUser}`)
            .reply(200, mockUserAPIReturn)

        const details = await getUserDetails({
            github: mockGithub,
            username: mockUser,
        })
        expect(details).toMatchSnapshot()
    })

    test('throws error when user does not exists', async () => {
        const mockNoUser = 'no-such-user'

        nock('https://api.github.com')
            .get(`/users/${mockNoUser}`)
            .reply(404, {
                message: 'Not Found',
                documentation_url:
                    'https://developer.github.com/v3/users/#get-a-single-user',
            })

        const error = await rejectionOf(
            getUserDetails({ github: mockGithub, username: mockNoUser }),
        )
        expect(error instanceof UserNotFoundError).toBeTruthy()
        expect(error.message).toMatchSnapshot()
    })

    test('throws error on 500', async () => {
        const mockUser = 'mock-user'

        nock('https://api.github.com')
            .get(`/users/${mockUser}`)
            .reply(500)

        const error = await rejectionOf(
            getUserDetails({ github: mockGithub, username: mockUser }),
        )
        expect(error instanceof Error).toBeTruthy()
    })
})
