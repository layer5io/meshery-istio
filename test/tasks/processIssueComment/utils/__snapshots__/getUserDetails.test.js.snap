// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get User Details gets user details when user exists 1`] = `
Object {
  "avatar_url": "https://avatars2.githubusercontent.com/u/3534236?v=4",
  "name": "Jake Bolam",
  "profile": "https://jakebolam.com",
}
`;

exports[`Get User Details throws error when user does not exists 1`] = `"Could not find the user \`no-such-user\` on github."`;
